sdk/eventhub/azure-eventhubs/api_review.py

```diff
@@ -0,0 +1,230 @@
+
+class EventHubProducerClient(ClientBaseAsync):
+
+    eh_name  # type: str
+
+    def __init__(
+            self,
+            host: str,
+            event_hub_path: str,
+            credential: TokenCredential,
+            **kwargs
+    ) -> None:
+
+    async def __aenter__(self):
+
+    async def __aexit__(self, *args):
+
+    @classmethod
+    def from_connection_string(
+        cls, conn_str: str,
+        *,
+        event_hub_path: str = None,
+        logging_enable: bool = False,
+        http_proxy: dict = None,
+        auth_timeout: float = 60,
+        user_agent: str = None,
+        retry_total: int = 3,
+        transport_type=None) -> EventHubProducerClient:
+
+    async def send(
```

Comment:
Agree to rename this to `send_batch` like JS.
Also moving the `partition_key` and `partition_id` parameters to the `create_batch` function.